src/libs/tmdk/e2e-tests/basic-e2e/basic-e2e.test.ts

```diff
@@ -0,0 +1,492 @@
+  twinMakerPermissionPolicy,
+  twinMakerPermissionPolicySuffix,
+} from "./basic-e2e-iam";
+import {
+  ComponentTypeSummary,
+  GetComponentTypeCommandOutput,
+} from "@aws-sdk/client-iottwinmaker";
+import { DeleteObjectCommand, PutObjectCommand } from "@aws-sdk/client-s3";
+import * as path from "path";
+import { EntitySummary } from "@aws-sdk/client-iottwinmaker/dist-types/models/models_0";
+import { delay } from "../../src/lib/utils";
+import prompts = require("prompts");
+
+// Ensure nuke always operates
+prompts.inject(["Y", "Y"]);
+
+test("basic e2e test", async () => {
+  console.log("//////      BEGIN BASIC E2E TEST     //////");
+  let argv2;
+
+  // 0. Get account info
+  console.log("Initializing AWS client.");
+  initDefaultAwsClients({ region: `${constants.region}` });
+  const accountId = (await aws().getCurrentIdentity())["accountId"];
+  const twinMakerRoleName = constants.workspaceId + "-" + accountId + "-role";
+  const twinMakerPolicyName =
+    twinMakerRoleName + twinMakerPermissionPolicySuffix;
+  const twinMakerRoleArn =
+    "arn:aws:iam::" + accountId + ":role/" + twinMakerRoleName;
+  const workspaceS3BucketName = constants.workspaceId + "-" + accountId;
+  const workspaceS3BucketArn = "arn:aws:s3:::" + workspaceS3BucketName;
+  const scene1S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.scene1FileName;
+  const model1S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.model1FileName;
+  const model2S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.model2FileName;
+
+  // 1. Clean up pre-existing resources, if any
+  console.log(
+    "Deleting test IAM role, s3 bucket, and workspace if they exist."
+  );
+  try {
+    await aws().tm.getWorkspace({ workspaceId: constants.workspaceId });
+    console.log("Workspace exists, nuking it.");
+    argv2 = {
+      _: ["nuke"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": constants.workspaceId,
+    } as Arguments<nuke.Options>;
+    await nuke.handler(argv2);
+    await aws().tm.deleteWorkspace({ workspaceId: constants.workspaceId });
+  } catch (e) {
+    console.log("Workspace does not exist, as expected.");
+  }
+  try {
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.model1FileName,
+      })
+    );
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.model2FileName,
+      })
+    );
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.scene1FileName,
+      })
+    );
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.scene2FileName,
+      })
+    );
+    await aws().s3.deleteBucket({ Bucket: workspaceS3BucketName });
+  } catch (e) {
+    console.log("S3 bucket does not exist, as expected.");
```

Comment:
do we need to handle the case where the first file doesn't exist but the rest do?

Reply:
No, s3 delete object commands do not throw exception if the object does not exist